### PR TITLE
feat(eventing): init with optional spawn function

### PR DESCRIPTION
**Pull request by datacore-vvarakantham**
With this code, we can use task spawning functionality in data plane

### CODE REVIEW

File: event-publisher/src/event_handler.rs [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
         initilize_source_component(service_name);
         let layer = EventLayer::new(send);
-        tokio::spawn(async move {
+        let publisher_future = Box::pin(async move {
             MbusPublisher::run(&mbus_url, recv).await;
-            // TODO handle the closed channel situation.
         });
+        match spawn_option {
+            Some(spawn_fn) => {
+                // Spawn the publisher on the tokio-runtime within mayastor.
```
Comment (tiagolobocastro) [inline event-publisher/src/event_handler.rs]:
```suggestion
                // Spawn the publisher on the spawner specified in the args.
```

File: event-publisher/src/event_handler.rs [hunk 2]
```diff
@@ -1,22 +1,37 @@
...
+                spawn_fn(publisher_future);
+            }
+            None => {
+                tokio::spawn(publisher_future);
+            } // TODO handle the closed channel situation.
```
Comment (tiagolobocastro) [inline event-publisher/src/event_handler.rs]: Should this comment be here on the the publisher_future above?

File: event-publisher/src/event_handler.rs [hunk 3]
```diff
@@ -1,22 +1,37 @@
...
 
 /// Event handle.
-pub struct EventHandle {}
+pub struct EventHandle<T> {
+    __phantom: PhantomData<T>,
+}
 
-impl EventHandle {
+impl<T> EventHandle<T>
+where
```
Comment (tiagolobocastro) [inline event-publisher/src/event_handler.rs]: You don't need to add T for the struct btw, I think can just add it to the init function

**Review by tiagolobocastro — APPROVED** [review]

**Review by tiagolobocastro — COMMENTED** [review]
rename PR and commit to simply `feat(eventing): init with optional spawn function`

File: event-publisher/src/event_handler.rs (hunk 2 above)
Comment (datacore-vvarakantham) [reply event-publisher/src/event_handler.rs]: No. It should be for publisher_future. Will move it.

**Review by sinhaashish — APPROVED** [review]
LGTM

**Pull request merged**